internal/metrics/proxy_metrics.go

```diff
@@ -0,0 +1,290 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"github.com/milvus-io/milvus/internal/util/typeutil"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	// ProxyDmlChannelTimeTick counts the time tick value of dml channels
+	ProxyDmlChannelTimeTick = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.ProxyRole,
+			Name:      "dml_channels_time_tick",
+			Help:      "Time tick of dml channels",
+		}, []string{"node_id", "pchan"})
+
+	// ProxySearchCount record the number of times search succeeded or failed.
+	ProxySearchCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.ProxyRole,
+			Name:      "search_counter",
+			Help:      "The number of times search succeeded or failed",
+		}, []string{"node_id", "collection", "type", "status"})
+
+	// ProxyInsertCount record the number of times insert succeeded or failed.
+	ProxyInsertCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.ProxyRole,
+			Name:      "insert_counter",
+			Help:      "The number of times insert succeeded or failed",
+		}, []string{"node_id", "collection", "status"})
+
+	// ProxySearchVectors record the number of vectors search successfully.
+	ProxySearchVectors = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.ProxyRole,
+			Name:      "search_vectors",
+			Help:      "The number of vectors search successfully",
+		}, []string{"node_id", "collection", "type"})
+
+	// ProxyInsertVectors record the number of vectors insert successfully.
+	ProxyInsertVectors = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.ProxyRole,
+			Name:      "insert_vectors",
+			Help:      "The number of vectors insert successfully",
+		}, []string{"node_id", "collection"})
+
+	// ProxyLinkedSDKs record The number of SDK linked proxy.
+	// TODO: how to know when sdk disconnect?
+	ProxyLinkedSDKs = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.ProxyRole,
+			Name:      "linked_SDK_numbers",
+			Help:      "The number of SDK linked proxy",
+		}, []string{"node_id"})
+
+	// ProxySearchLatency record the latency of search successfully.
+	ProxySearchLatency = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.ProxyRole,
+			Name:      "search_latency",
+			Help:      "The latency of search successfully",
+			Buckets:   []float64{10, 50, 100, 500, 1000}, // unit: ms
```

Comment:
Is there any reason why we choose 10, 50, 100, 500, 1000? Maybe we should record more bucket. setcd server metrics always record 14 buckets.

Reply:
That sounds great, I will add more buckets for Histogram.